### DISCUSSION TITLE
should we know TypeScript?

### DISCUSSION THREAD

**Discussion by pishguy** [adonisjs/core#1729] [Help] [answered]
in new version of the framework, used typescript and i want to know, we should be know typescript?

**Comment by dajpes** [comment]
I only know Js and I can more or less handle typescript in adonis v5

**Comment by thecodemill** [comment]
This is also a significant barrier for entry with my dev team as well. We'd really like to give Adonis a spin for a new project. But with v5 around the corner we probably wouldn't enter with v4. And we actively avoid Typescript too (multiple reasons; business and technical), which sadly leaves us looking for alternatives :(

**Comment by thetutlage** [comment, marked as answer]
Yes knowing Typescript is a prerequisite as you will author your applications in Typescript

**Reply by McSneaky** [reply]
What I'd like to add tho is that all JS is fully valid TS

Only thing that you have to use from TS in V5 is decorators at models (which are documented, you can copy-paste them)

So I wouldn't really say that you need to be TS guru or anything. Knowing it would be good (especially when working on something bigger) but can also learn it on the fly
